### PR TITLE
update for major version 5.0

### PR DESCRIPTION
**Pull request by cdhoffmann**
* Bumps the minimum deployment target to iOS 12.0
* Restrict podspec dependencies to >= 5.0.0 AND < 6.0.0
* Updates CI to use Xcode 15 and iOS 17
* removes deprecated -swift-version flag and .swift_version file for cocoapods (uses s.swift_version in podspec now)
* Use Codecov github action instead of bash uploader which is deprecated

### CODE REVIEW

File: AEPUserProfile/Sources/UserProfile.swift [hunk 1]
```diff
@@ -16,7 +16,7 @@ import Foundation
 
 @objc(AEPMobileUserProfile)
 public class UserProfile: NSObject, Extension {
-    internal static let LOG_TAG = "UserProfile"
```
Comment (praveek) [inline AEPUserProfile/Sources/UserProfile.swift:19]: This can be private.

**Review by praveek — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ The Adobe Experience Platform UserProfile Mobile Extension is an extension for t
 To learn more about this extension, read [Adobe Experience Platform Profile Mobile Extension](https://aep-sdks.gitbook.io/docs/v/AEP-Edge-Docs/).
 
 ## Requirements
-- Xcode 14.1 (or newer)
+- Xcode 15 (or newer)
 - Swift 5.1 (or newer)
```
Comment (swarna04) [inline README.md:16]: Requires doc update for swift version?

File: AEPUserProfile.podspec [hunk 3]
```diff
@@ -10,10 +10,10 @@ Pod::Spec.new do |s|
   s.author           = "Adobe Experience Platform SDK Team"
   s.source           = { :git => "https://github.com/adobe/aepsdk-userprofile-ios", :tag => s.version.to_s }
 
-  s.ios.deployment_target = '11.0'
+  s.ios.deployment_target = '12.0'
   s.swift_version    = '5.1'
```
Comment (swarna04) [inline AEPUserProfile.podspec:14]: Should bump up the version here?

Comment (cdhoffmann) [reply AEPUserProfile.podspec:14]: @swarna04 The swift version is remaining the same.

**Pull request merged**